app/helpers/protocols_io_helper.rb

```diff
@@ -165,7 +165,13 @@ def close_open_html_tags(text)
   end
 
   def step_hash_null?(step_json)
-    step_json.dig(0, 'components', 0, 'component_type_id').nil?
+    return false unless step_json.dig(
+      0, 'components', 0, 'component_type_id'
+    ).nil?
+    return false unless step_json.dig(
+      0, 'components', '0', 'component_type_id'
+    ).nil?
+    true
```

Comment:
You can make it shorter:
`step_json.dig(0, 'components', 0, 'component_type_id').nil? ||`
  `step_json.dig(0, 'components', '0', 'component_type_id').nil?`